pages/_app.js

```diff
@@ -186,6 +190,48 @@ export default function App({
   function handleResetFilter() {
     setSearchTerm("");
   }
+  useEffect(() => {
+    if ("serviceWorker" in navigator && "PushManager" in window) {
+      window.addEventListener("beforeinstallprompt", (event) => {
+        event.preventDefault();
+        setDeferredPrompt(event);
+        setShowInstallButton(true);
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    window.addEventListener("beforeinstallprompt", (event) => {
+      event.preventDefault();
+      setDeferredPrompt(event);
+      setShowInstallButton(true);
+      setShowInstallPrompt(false);
+    });
+
+    window.addEventListener("appinstalled", () => {
+      setShowInstallPrompt(false);
+      setShowInstallButton(true);
+      console.log("App installed");
+      setDeferredPrompt(null);
+    });
+
+    return () => {
+      window.removeEventListener("beforeinstallprompt", () => {});
+      window.removeEventListener("appinstalled", () => {});
+    };
+  }, []);
+
+  const handleInstallClick = async () => {
```

Comment:
your other handlers are defined as functions, please use the same syntax for the same concepts (like functions inside your components). I mentioned this now multiple times, please make sure to double check your code before opening the PR.